src/Functions/randDistribution.cpp

```diff
@@ -93,7 +92,7 @@ struct ChiSquaredDistribution
 
     static void generate(Float64 degree_of_freedom, ColumnFloat64::Container & container)
     {
-        if (degree_of_freedom <= 0)
+        if (!container.empty() && degree_of_freedom <= 0)
```

Comment:
Why? The `degree_of_freedom` is parameter of distribution. We should check unconditionally.

Reply:
I've checked it and when we run for example randFisherF(1 + (rand() % 1), 2), some of the first calls of this generate)_, contain ColumnFloat64::Container which is empty. I've just added this check so we exclude this check, when 1 + (rand() % 1) isn't resolved yet (container is empty, and degree_of_freedom = 0).